kernel/uGraph.mli

```diff
@@ -9,63 +9,72 @@
 open Univ
 
 (** {6 Graphs of universes. } *)
+module Public : sig
 
 type t
-
 type universes = t
+[@@ocaml.deprecated "Use UGraph.t"]
+
+type 'a check_function = t -> 'a -> 'a -> bool
 
-type 'a check_function = universes -> 'a -> 'a -> bool
 val check_leq : Universe.t check_function
 val check_eq : Universe.t check_function
 val check_eq_level : Level.t check_function
 
-(** The empty graph of universes *)
-val empty_universes : universes
-
 (** The initial graph of universes: Prop < Set *)
-val initial_universes : universes
+val initial_universes : t
 
-val is_initial_universes : universes -> bool
+(** Check equality of instances w.r.t. a universe graph *)
+val check_eq_instances : Instance.t check_function
 
-val sort_universes : universes -> universes
+(** {6 ... } *)
+(** Merge of constraints in a universes graph.
+  The function [merge_constraints] merges a set of constraints in a given
+  universes graph. It raises the exception [UniverseInconsistency] if the
+  constraints are not satisfiable. *)
+
+val enforce_constraint : univ_constraint -> t -> t
+val merge_constraints : constraints -> t -> t
+
+val check_constraint  : t -> univ_constraint -> bool
+val check_constraints : constraints -> t -> bool
 
 (** Adds a universe to the graph, ensuring it is >= or > Set.
    @raises AlreadyDeclared if the level is already declared in the graph. *)
 
 exception AlreadyDeclared
 
-val add_universe : Level.t -> bool -> universes -> universes
+val add_universe : Level.t -> bool -> t -> t
 
-(** {6 ... } *)
-(** Merge of constraints in a universes graph.
-  The function [merge_constraints] merges a set of constraints in a given
-  universes graph. It raises the exception [UniverseInconsistency] if the
-  constraints are not satisfiable. *)
+(** {6 Pretty-printing of universes. } *)
 
-val enforce_constraint : univ_constraint -> universes -> universes
-val merge_constraints : constraints -> universes -> universes
+val pr_universes : (Level.t -> Pp.t) -> t -> Pp.t
 
-val constraints_of_universes : universes -> constraints
+end
 
-val check_constraint  : universes -> univ_constraint -> bool
-val check_constraints : constraints -> universes -> bool
+include module type of Public
 
-val check_eq_instances : Instance.t check_function
-(** Check equality of instances w.r.t. a universe graph *)
+module Internal : sig
+
+(** The empty graph of universes *)
+val empty_universes : t
+
+val is_initial_universes : t -> bool
```

Comment:
Why is this internal?

Reply:
No reason, maybe it was not exported in API? Bear with me, the universe API is the thing I'm less familiar with the whole codebase. Making this public.